### PR TITLE
Stream model deployer logs through CLI

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes

Implements a `zenml served-models logs` CLI command that streams the log contents of
model servers through the CLI, for easy access to the back-end logs. The command supports "following" the remote logs, "tailing" the last X lines of logs and also pretty-prints everything in the
logs by default, so the result is an improved version of the remote logs. Large files are also supported through the use of IO buffering, HTTP streaming and Python generators.

Example output:

Screenshot from 2022-05-03 12-42-27

Log streaming now works not only with the Seldon and MLflow model servers, but also with
every local daemon service. The base Service class now specifies this abstract method:

```python
    @abstractmethod
    def get_logs(
        self, follow: bool = False, tail: Optional[int] = None
    ) -> Generator[str, bool, None]:
``` 

TODO:
- [x] support multiple "log sources" for services that are implemented using multiple components, such as Kubernetes deployments
- [x] add init containers to the list of logs sources that can be retrieved for Seldon Core deployments

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/services/service.py [hunk 1]
```diff
@@ -174,6 +174,23 @@ def check_status(self) -> Tuple[ServiceState, str]:
             service status).
         """
 
+    @abstractmethod
```
Comment (schustmi) [inline src/zenml/services/service.py:177]: No request to change anything, but making sure that this was intended as a non-optional method that any service needs to implement?

**Review by schustmi — CHANGES_REQUESTED** [review]
Found one tiny type annotation that can be improved (but I had to look for a long time ;). Looks great otherwise!

File: src/zenml/services/service.py (hunk 1 above)
Comment (stefannica) [reply src/zenml/services/service.py:177]: Yes, I would consider this to be part of the service abstraction. If it cannot be implemented, then the subclass can just throw an exception or log a message saying so.

**Review by AlexejPenner — APPROVED** [review]
Looks good to me! :)

**Review by schustmi — APPROVED** [review]
LGTM!

**Pull request merged**